core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -497,17 +498,18 @@ protected void onModified() throws IOException {
     }
 
     /**
-     * Set of installed cluster nodes.
-     * <p>
-     * We use this field with copy-on-write semantics.
-     * This field has mutable list (to keep the serialization look clean),
-     * but it shall never be modified. Only new completely populated slave
-     * list can be set here.
-     * <p>
-     * The field name should be really {@code nodes}, but again the backward compatibility
-     * prevents us from renaming.
+     * Legacy store of the set of installed cluster nodes.
+     * @deprecated in favour of {@link Nodes}
+     */
+    @Deprecated
+    protected transient volatile NodeList slaves;
+
+    /**
+     * The holder of the set of installed cluster nodes.
+     *
+     * @since 1.FIXME
      */
-    protected volatile NodeList slaves;
```

Comment:
Note that this introduces a behavioral incompatibility: `Jenkins.getInstance().save()` no longer saves modifications to `Node`s, such as changes to `NodeProperty` state made by means other than the configuration screen.